alpha/apps/kaltura/lib/kMrssManager.php

```diff
@@ -244,7 +244,23 @@ public static function getEntryMrss(entry $entry, SimpleXMLElement $mrss = null,
 		$mrss = self::getEntryMrssXml($entry, $mrss, $mrssParams);
 		return $mrss->asXML();
 	}
-	
+
+	protected static function enforceHttpsIfRequired($partnerId, $url)
+	{
+		try
+		{
+			$partner = PartnerPeer::retrieveByPK($partnerId);
+			if ($partner && $partner->getEnforceHttpsApi())
+			{
+				$url = str_replace('http://', 'https://', $url);
```

Comment:
In case the URL contains http:// more than once, better limit it to the first occurance 
  if (strpos($url, 'http://') === 0) 
 {
      $url = 'https://' . substr($url, 7);
  }